pkg/build/builder/docker.go

```diff
@@ -416,6 +426,49 @@ func traverseAST(cmd string, node *parser.Node) int {
 	return index
 }
 
+// insertEnvAfterFrom inserts an ENV instruction with the environment variables
+// from env after every FROM instruction in the given Dockerfile instructions
+// and returns a new Dockerfile.
+func insertEnvAfterFrom(instructions []byte, env []kapi.EnvVar) ([]byte, error) {
+	if len(env) == 0 {
+		return instructions, nil
+	}
+	node, err := parser.Parse(bytes.NewReader(instructions))
+	if err != nil {
+		return nil, err
+	}
+
+	// buildEnv is an ENV Dockerfile instruction to define all environment
+	// variables from env.
+	buildEnv := []byte("ENV")
```

Comment:
Can you use dockercmd.Env instead? You'll probably have to uppercase it first but I think it's much better than using plain strings. WDYT?